src/main/java/com/oltpbenchmark/benchmarks/auctionmark/AuctionMarkWorker.java

```diff
@@ -447,8 +447,24 @@ protected boolean executeCloseAuctions(Connection conn, CloseAuctions proc) thro
         Timestamp startTime = profile.getLastCloseAuctionsTime();
         Timestamp endTime = profile.updateAndGetLastCloseAuctionsTime();
 
-        List<Object[]> results = proc.run(conn, benchmarkTimes, startTime, endTime);
-
+        List<Object[]> results = null;
+        boolean done = false;
+        // Retry here the close auction procedure if it fails due to a transactional conflict, since
+        // the original submitted procedured has a different type.
+        while (!done) {
+            try {
+                results = proc.run(conn, benchmarkTimes, startTime, endTime);
+                done = true;
+            }
```

Comment:
I actually don't think we want to do this because we can't keep track of the # of times that a txn is submitted. So this can cause BenchBase to exceed the defined submission rate. Also, by retrying inside of the benchmark worker we can't keep track of the # failed txns.

Reply:
I also agree that this is not the best approach. I also thought about adding an update method to the "TransactionType" class, so we can update the original "txnType" when we switch it in the "executeWork" function, making the worker deal with retries. This also would mean that the CloseAuctions procedure would be correctly logged in the results, since right now, as far as the worker knows, no CloseAuctions procedure is executed.